x-pack/plugins/enterprise_search/server/utils/elasticsearch_error_handler.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { RequestHandler } from '@kbn/core/server';
+import { Logger } from '@kbn/core/server';
+
+import { i18n } from '@kbn/i18n';
+
+import { ErrorCode } from '../../common/types/error_codes';
+
+import { createError, EnterpriseSearchError } from './create_error';
+import { isUnauthorizedException } from './identify_exceptions';
+
+export function elasticsearchErrorHandler<ContextType, RequestType, ResponseType>(
+  log: Logger,
+  requestHandler: RequestHandler<ContextType, RequestType, ResponseType>
+): RequestHandler<ContextType, RequestType, ResponseType> {
+  return async (context, request, response) => {
+    try {
+      return await requestHandler(context, request, response);
+    } catch (error) {
+      let enterpriseSearchError: EnterpriseSearchError | undefined;
+
+      if (isUnauthorizedException(error)) {
+        enterpriseSearchError = {
+          errorCode: ErrorCode.UNAUTHORIZED,
+          message: i18n.translate('xpack.enterpriseSearch.server.routes.unauthorizedError', {
+            defaultMessage: 'You do not have sufficient permissions.',
+          }),
+          statusCode: 403,
+        };
+      } else {
+        enterpriseSearchError = {
+          errorCode: ErrorCode.UNCAUGHT_EXCEPTION,
+          message: i18n.translate('xpack.enterpriseSearch.server.routes.uncaughtExceptionError', {
+            defaultMessage: 'Enterprise Search encountered an error.',
+          }),
+          statusCode: 502,
+        };
+      }
+
+      if (typeof enterpriseSearchError !== 'undefined') {
```

Comment:
question: what's the advantage of doing `typeof !== 'undefined'` over just checking `enterpriseSearchError !== undefined` (or even just `!enterpriseSearchError` in this case)?

Reply:
Hmmm I was taught to use typeof over strict equality a decade ago but now that I'm googling it I'm not really seeing any reason in favor of it